plugins/search/src/alpha.tsx

```diff
@@ -0,0 +1,496 @@
+
+                        const names = items.map(entity => entity.metadata.name);
+                        names.sort();
+                        return names;
+                      }}
+                    />
+                  </Grid>
+                )}
+                <Grid item xs={2}>
+                  <SearchFilter.Select
+                    className={classes.filter}
+                    label="Kind"
+                    name="kind"
+                    values={['Component', 'Template']}
+                  />
+                </Grid>
+                <Grid item xs={2}>
+                  <SearchFilter.Select
+                    className={classes.filter}
+                    label="Lifecycle"
+                    name="lifecycle"
+                    values={['experimental', 'production']}
+                  />
+                </Grid>
+                <Grid
+                  item
+                  xs={types.includes('techdocs') ? 6 : 8}
+                  container
+                  direction="row-reverse"
+                  justifyContent="flex-start"
+                  alignItems="center"
+                >
+                  <Grid item>
+                    <Button
+                      className={classes.button}
+                      color="primary"
+                      endIcon={<ArrowForwardIcon />}
+                      onClick={handleSearchBarSubmit}
+                      disableRipple
+                    >
+                      View Full Results
+                    </Button>
+                  </Grid>
+                </Grid>
+              </Grid>
+              <Grid item xs>
+                <SearchResults>
+                  {({ results }) =>
+                    results.map((result, index) => {
+                      const SearchResultListItem =
+                        getResultItemComponent(result);
+                      return (
+                        <SearchResultListItem
+                          key={index}
+                          rank={result.rank}
+                          result={result.document}
+                          noTrack={config.noTrack}
+                        />
+                      );
+                    })
+                  }
+                </SearchResults>
+              </Grid>
+            </Grid>
+          </DialogContent>
+          <DialogActions className={classes.dialogActionsContainer}>
+            <Grid container direction="row">
+              <Grid item xs={12}>
+                <SearchResultPager />
+              </Grid>
+            </Grid>
+          </DialogActions>
+        </>
+      );
+    };
+
+    bind({
+      element: <Component />,
+    });
+  },
+});
+
+/** @alpha */
+export const SearchNavItem = createNavItemExtension({
+  id: 'plugin.search.nav.index',
+  routeRef: searchRouteRef,
+  title: 'Search',
+  icon: SearchIcon,
+  group: async ({ inputs }) => (
+    <SidebarSearchModal>{() => inputs.modal.element}</SidebarSearchModal>
+  ),
```

Comment:
Hmm, not thrilled about this tbh, feels very leaky. What is the "group" here really and how is it supposed to be used?

I think it might be best to leave the search field in the sidebar fairly hardcoded for now. We could for example go with a search modal input on the core nav, where when it's registered causes the search item to show up. I don't think that's perfect either but it makes sure we keep the nav items very slim for now

Reply:
Thanks for commenting on it, I didn't feel strongly that it was the best way either 🙏🏻 .
I removed the modal for now and added a comment for us to work on a solution later on.
Does it sound good?
https://github.com/backstage/backstage/pull/20169/commits/09ba8b0c955af2f4dfd053648b5ce5d6086a8327